pages/bookshelf/series/_id.vue

```diff
@@ -19,10 +23,167 @@ export default {
     }
   },
   data() {
-    return {}
+    return {
+      startingDownload: false,
+      mediaType: 'book',
+      booksPerFetch: 20,
+      books: 0,
+      numFiles: 0,
+      numAudioFiles: 0,
+      libraryIds: []
+    }
+  },
+  mixins: [cellularPermissionHelpers],
+  computed: {
+    isIos() {
+      return this.$platform === 'ios'
+    }
+  },
+  methods: {
+    async downloadSerieClick() {
+      console.log('Download Series clicked')
+      if (this.startingDownload) return
+
+      const hasPermission = await this.checkCellularPermission('download')
+      if (!hasPermission) return
+
+      this.startingDownload = true
+      setTimeout(() => {
+        this.startingDownload = false
+      }, 1000)
+
+      await this.$hapticsImpact()
+      this.download()
+    },
+    buildSearchParams() {
+      let searchParams = new URLSearchParams()
+      searchParams.set('filter', `series.${this.$encode(this.seriesId)}`)
+      return searchParams.toString()
+    },
+    async fetchSerieEntities(page) {
+      const startIndex = page * this.booksPerFetch
+
+      this.currentSFQueryString = this.buildSearchParams()
+
+      const entityPath = `items`
+      const sfQueryString = this.currentSFQueryString ? this.currentSFQueryString + '&' : ''
+      const fullQueryString = `?${sfQueryString}limit=${this.booksPerFetch}&page=${page}&minified=1&include=rssfeed,numEpisodesIncomplete`
+
+      const payload = await this.$nativeHttp.get(`/api/libraries/${this.series.libraryId}/${entityPath}${fullQueryString}`).catch((error) => {
+        console.error('failed to fetch books', error)
+        return null
+      })
+
+      if (payload && payload.results) {
+        console.log('Received payload', payload)
+        this.books = payload.total
+
+        for (let i = 0; i < payload.results.length; i++) {
+          console.log(payload.results[i].numFiles)
+          if (!(await this.$db.getLocalLibraryItem(`local_${payload.results[i].id}`))) {
+            this.numFiles += payload.results[i].numFiles
```

Comment:
This is including any file which is not an audio file as ebook files. Downloading a series which has no ebook files shows multiple ebook files as being downloaded due to cover images.

Reply:
Changing confirmation message to suggested resolved this